core/src/main/java/hudson/WebAppMain.java

```diff
@@ -262,6 +264,25 @@ public void joinInit() throws InterruptedException {
     }
 
     /**
+     * Creates the plugin manager to use, if customized.
+     * @return The plugin manager to use or {@code null} to use the default one.
+     */
+    private PluginManager createPluginManager(ServletContext context, File rootDir) {
+        final String pmClassName = context.getInitParameter(PluginManager.class.getName() + ".className");
+        if (pmClassName == null) {
+            return null;
+        }
+        try {
+            final Class<? extends PluginManager> klass = Class.forName(pmClassName).asSubclass(PluginManager.class);
```

Comment:
But in any case, whoever is returning `null` would be breaking the class contract...